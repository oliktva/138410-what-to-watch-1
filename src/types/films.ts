import PropTypes from 'prop-types';

export const filmPropTypes = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  video: PropTypes.string.isRequired
});

export const filmsPropTypes = PropTypes.arrayOf(filmPropTypes.isRequired);

export interface FilmProps {
  name: string;
  img: string;
  video: string;
  id: number;
}
