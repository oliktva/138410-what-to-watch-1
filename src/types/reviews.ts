export interface ReviewProps {
  comment: string;
  date: string;
  id: number;
  rating: number;
  user: {
    id: number;
    name: string;
  };
}
