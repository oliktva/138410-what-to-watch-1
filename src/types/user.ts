import PropTypes from 'prop-types';

export const UserPropTypes = PropTypes.shape({
  id: PropTypes.number,
  email: PropTypes.string,
  name: PropTypes.string,
  avatarUrl: PropTypes.string,
});

export interface UserProps {
  id: number | null;
  email: string | null;
  name: string | null;
  avatarUrl: string | null;
}
