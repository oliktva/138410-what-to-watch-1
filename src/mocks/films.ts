import {FilmProps} from "src/types/films";

export const films: FilmProps[] = [{
  id: 1,
  name: `Fantastic Beasts`,
  img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 2,
  name: `Bohemian Rhapsody`,
  img: `img/bohemian-rhapsody.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 3,
  name: `Macbeth`,
  img: `img/macbeth.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 4,
  name: `Aviator`,
  img: `img/aviator.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 5,
  name: `We need to talk about Kevin`,
  img: `img/we-need-to-talk-about-kevin.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 6,
  name: `What We Do in the Shadows`,
  img: `img/what-we-do-in-the-shadows.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 7,
  name: `Revenant`,
  img: `img/revenant.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 8,
  name: `Johnny English`,
  img: `img/johnny-english.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 9,
  name: `Shutter Island`,
  img: `img/shutter-island.jpg`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
}, {
  id: 10,
  name: `Pulp Fiction`,
  img: `img/pulp-fiction.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 11,
  name: `No Country for Old Men`,
  img: `img/no-country-for-old-men.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 12,
  name: `Snatch`,
  img: `img/snatch.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 13,
  name: `Moonrise Kingdom`,
  img: `img/moonrise-kingdom.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 14,
  name: `Seven Years in Tibet`,
  img: `img/seven-years-in-tibet.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 15,
  name: `Midnight Special`,
  img: `img/midnight-special.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 16,
  name: `War of the Worlds`,
  img: `img/war-of-the-worlds.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 17,
  name: `Dardjeeling Limited`,
  img: `img/dardjeeling-limited.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 18,
  name: `Orlando`,
  img: `img/orlando.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 19,
  name: `Mindhunter`,
  img: `img/mindhunter.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}, {
  id: 20,
  name: `Midnight Special`,
  img: `img/midnight-special.jpg`,
  video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
}];
