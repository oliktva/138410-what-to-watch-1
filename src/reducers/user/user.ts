import {Action as ReduxAction} from 'redux';
import {AxiosResponse, AxiosError, AxiosInstance} from 'axios';
import {toCamel} from 'convert-keys';

import {UserProps} from 'src/types/user';
import {ThunkDispatch, ThunkAction, State as AppState} from 'src/types/reducer';

export const TOGGLE_AUTHORIZATION_REQUIRED = `TOGGLE_AUTHORIZATION_REQUIRED`;
export const LOG_IN_USER = `LOG_IN_USER`;
export const LOG_IN_USER_ERROR = `LOG_IN_USER_ERROR`;

export interface State {
  isAuthorizationRequired: boolean;
  id?: number;
  email?: string;
  name?: string;
  avatarUrl?: string;
  error?: string;
}

interface LogInUserAction extends ReduxAction {
  type: typeof LOG_IN_USER;
  payload: UserProps;
}

interface LogInUserErrorAction extends ReduxAction {
  type: typeof LOG_IN_USER_ERROR;
  payload: string;
}

export type Action = LogInUserAction | LogInUserErrorAction;

export const initialState: State = {
  isAuthorizationRequired: false,
  id: undefined,
  email: undefined,
  name: undefined,
  avatarUrl: undefined,
  error: undefined
};

export const ActionCreator = {
  logInUser: (user: UserProps): LogInUserAction => {
    return {
      type: LOG_IN_USER,
      payload: user
    };
  },

  logInUserError: (error: string): LogInUserErrorAction => {
    return {
      type: LOG_IN_USER_ERROR,
      payload: error
    };
  }
};

export const Operation = {
  loadUser: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/login`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<UserProps>(response.data);

          dispatch(ActionCreator.logInUser(data));
        }).catch((): void => {
          // don't care
        });
    };
  },

  logInUser: (email: string, password: string): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/login`, {
        email,
        password
      }).then((response: AxiosResponse<Record<string, any>>): void => {
        const data = toCamel<UserProps>(response.data);

        dispatch(ActionCreator.logInUser(data));
      }).catch((error: AxiosError): void => {
        dispatch(ActionCreator.logInUserError(error.message));
      });
    };
  }
};

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case LOG_IN_USER:
      return {
        ...state,
        ...action.payload,
        error: undefined
      };

    case LOG_IN_USER_ERROR:
      return {
        ...state,
        error: action.payload
      };

    default:
      return state;
  }
};

export default reducer;
