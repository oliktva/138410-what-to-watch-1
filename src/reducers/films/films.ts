import {Action as ReduxAction} from 'redux';
import {ThunkDispatch as ReduxThunkDispatch, ThunkAction as ReduxThunkAction} from 'redux-thunk';
import {AxiosResponse, AxiosInstance} from 'axios';
import {toCamel} from 'convert-keys';

import {GenreProps} from 'src/types/genres';
import {FilmProps} from 'src/types/films';
import {ReviewProps} from 'src/types/reviews';
import {ThunkDispatch, ThunkAction, State as AppState} from 'src/types/reducer';

export const SET_GENRE = `SET_GENRE`;
export const LOAD_FILMS_SUCCESS = `LOAD_FILMS_SUCCESS`;
export const LOAD_FAVORITE_SUCCESS = `LOAD_FAVORITE_SUCCESS`;
export const UPDATE_FILM = `UPDATE_FILM`;
export const LOAD_PROMO = `LOAD_PROMO`;
export const LOAD_REVIEWS = `LOAD_REVIEWS`;

const ALL_GENRES = `All genres`;

export interface State {
  genre: GenreProps;
  items: FilmProps[];
  favorite: FilmProps[];
  promo: FilmProps | null;
  reviews: {[key: number]: ReviewProps[]} | null;
}

interface SetFilterAction extends ReduxAction {
  type: typeof SET_GENRE;
  payload: GenreProps;
}

interface LoadFilmsAction extends ReduxAction {
  type: typeof LOAD_FILMS_SUCCESS;
  payload: FilmProps[];
}

interface LoadFavoritesAction extends ReduxAction {
  type: typeof LOAD_FAVORITE_SUCCESS;
  payload: FilmProps[];
}

interface UpdateFilmAction extends ReduxAction {
  type: typeof UPDATE_FILM;
  payload: FilmProps;
}

interface LoadFilmPromoAction extends ReduxAction {
  type: typeof LOAD_PROMO;
  payload: FilmProps;
}

interface LoadReviewsAction extends ReduxAction {
  type: typeof LOAD_REVIEWS;
  payload: {id: number; reviews: ReviewProps[]};
}

export type Action =
  SetFilterAction | LoadFilmsAction | LoadFavoritesAction | UpdateFilmAction | LoadFilmPromoAction | LoadReviewsAction;

export type ThunkDispatch = ReduxThunkDispatch<State, AxiosInstance, Action>;
export type ThunkAction = ReduxThunkAction<Promise<void>, State, AxiosInstance, Action>;

export const initialState: State = {
  genre: ALL_GENRES,
  items: [],
  favorite: [],
  promo: null,
  reviews: null
};

const updateFilm = (film: FilmProps, filmsList: FilmProps[]): FilmProps[] =>
  filmsList.map((f: FilmProps): FilmProps => (f.id === film.id ? film : f));

export const ActionCreator = {
  setFilterByGenre: (genre: GenreProps): SetFilterAction => {
    return {
      type: SET_GENRE,
      payload: genre
    };
  },

  loadFilms: (films: FilmProps[]): LoadFilmsAction => {
    return {
      type: LOAD_FILMS_SUCCESS,
      payload: films
    };
  },

  loadFavorite: (films: FilmProps[]): LoadFavoritesAction => {
    return {
      type: LOAD_FAVORITE_SUCCESS,
      payload: films
    };
  },

  updateFilm: (film: FilmProps): UpdateFilmAction => {
    return {
      type: UPDATE_FILM,
      payload: film
    };
  },

  loadPromo: (film: FilmProps): LoadFilmPromoAction => {
    return {
      type: LOAD_PROMO,
      payload: film
    };
  },

  loadReviews: (id: number, reviews: ReviewProps[]): LoadReviewsAction => {
    return {
      type: LOAD_REVIEWS,
      payload: {id, reviews}
    };
  }
};

export const Operation = {
  loadFilms: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/films`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((f): FilmProps => toCamel<FilmProps>(f));

          dispatch(ActionCreator.loadFilms(data));
        });
    };
  },

  loadFavorite: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/favorite`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((f): FilmProps => toCamel<FilmProps>(f));

          dispatch(ActionCreator.loadFavorite(data));
        });
    };
  },

  addToFavorites: (filmId: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/favorite/${filmId}/1`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.updateFilm(data));
        });
    };
  },

  removeFromFavorites: (filmId: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/favorite/${filmId}/0`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.updateFilm(data));
        });
    };
  },

  loadPromo: (): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/films/promo`)
        .then((response: AxiosResponse<Record<string, any>>): void => {
          const data = toCamel<FilmProps>(response.data);

          dispatch(ActionCreator.loadPromo(data));
        });
    };
  },

  loadReviews: (id: number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.get(`/comments/${id}`)
        .then((response: AxiosResponse<Record<string, any>[]>): void => {
          const data = response.data.map((r): ReviewProps => toCamel<ReviewProps>(r));

          dispatch(ActionCreator.loadReviews(id, data));
        });
    };
  },

  addReview: (id: number, reviewText: string, rating: string | number): ThunkAction => {
    return (dispatch: ThunkDispatch, _getState: () => AppState, api: AxiosInstance): Promise<void> => {
      return api.post(`/comments/${id}`, {
        comment: reviewText,
        rating: typeof rating === `string` ? parseInt(rating, 10) : rating
      }).then((response: AxiosResponse<Record<string, any>[]>): void => {
        const data = response.data.map((r): ReviewProps => toCamel<ReviewProps>(r));

        dispatch(ActionCreator.loadReviews(id, data));
      });
    };
  }
};

const reducer = (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case LOAD_FILMS_SUCCESS:
      return {
        ...state,
        items: action.payload
      };

    case SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };

    case LOAD_FAVORITE_SUCCESS:
      return {
        ...state,
        favorite: action.payload
      };

    case UPDATE_FILM:
      return {
        ...state,
        promo: state.promo ? updateFilm(action.payload, [state.promo])[0] : state.promo,
        items: updateFilm(action.payload, state.items),
        favorite: updateFilm(action.payload, state.favorite)
      };

    case LOAD_PROMO: {
      return {
        ...state,
        promo: action.payload
      };
    }

    case LOAD_REVIEWS: {
      return {
        ...state,
        reviews: {
          ...state.reviews,
          [action.payload.id]: action.payload.reviews
        }
      };
    }

    default:
      return state;
  }
};

export default reducer;
