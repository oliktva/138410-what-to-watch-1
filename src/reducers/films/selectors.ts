import {createSelector} from 'reselect';

import Namespaces from 'src/reducers/namespaces';

import {State} from 'src/types/reducer';
import {FilmProps} from 'src/types/films';
import {GenreProps} from 'src/types/genres';

const ALL_GENRES = `All genres`;

export const getFilms = (state: State): FilmProps[] => {
  return state[Namespaces.FILMS].items;
};

export const getGenre = (state: State): GenreProps => {
  return state[Namespaces.FILMS].genre;
};

export const getFilmsByGenre = createSelector(
  getFilms,
  getGenre,
  (films: FilmProps[], genre: GenreProps): FilmProps[] => {
    return genre === ALL_GENRES ?
      films :
      films.filter((f: FilmProps): boolean => f.genre === genre);
  }
);

export const getGenres = createSelector(
  getFilms,
  (films: FilmProps[]): GenreProps[] => {
    const filmsGenres = films.map((f: FilmProps): GenreProps => f.genre)
      .reduce((result: GenreProps[], genre: GenreProps): GenreProps[] => {
        if (!result.includes(genre)) {
          result.push(genre);
        }

        return result;
      }, []);

    return [ALL_GENRES, ...filmsGenres];
  }
);
