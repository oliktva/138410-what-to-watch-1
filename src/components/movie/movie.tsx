import React, {PureComponent, ReactElement} from 'react';
import {connect, MapStateToProps, MapDispatchToProps} from 'react-redux';
import {RouteComponentProps} from 'react-router';

import {getGenresFilms} from 'src/helpers/genre-helpers';
import {Operation} from 'src/reducers/films/films';
import {getFilm, getReviews, getFilms} from 'src/reducers/films/selectors';
import {getUser} from 'src/reducers/user/selectors';

import PageWrapper from 'src/components/page-wrapper/page-wrapper';
import Header from 'src/components/header/header';
import FullMovieCard from 'src/components/full-movie-card/full-movie-card';

import {State, ThunkDispatch} from 'src/types/reducer';
import {FilmProps} from 'src/types/films';
import {ReviewProps} from 'src/types/reviews';
import {UserProps} from 'src/types/user';

interface StateProps {
  film: FilmProps | null;
  films: FilmProps[];
  reviews: ReviewProps[];
  user: UserProps;
}

interface DispatchProps {
  loadFilms: () => void;
  loadReviews: (id: number) => void;
}

type OwnProps = RouteComponentProps<{id: string}>;

type Props = StateProps & DispatchProps & OwnProps;

class Movie extends PureComponent<Props> {
  public componentDidMount(): void {
    const {loadFilms, loadReviews, match: {params: {id}}} = this.props;

    Promise.all([loadFilms(), loadReviews(parseInt(id, 10))]);
  }

  public render(): ReactElement | null {
    const {film, films, reviews, user} = this.props;

    if (film === null) {
      return null;
    }

    const relatedFilms = getGenresFilms(films, film.genre)
      .filter((f: FilmProps): boolean => f.id !== film.id)
      .slice(0, 4);

    const sortedReviews = reviews.sort((r1: ReviewProps, r2: ReviewProps): number => {
      if (r1.date < r2.date) {
        return 1;
      } else if (r1.date > r2.date) {
        return -1;
      }

      return 0;
    });

    return (
      <PageWrapper>
        <FullMovieCard
          header={<Header user={user} className="movie-card__head" />}
          film={film}
          reviews={sortedReviews}
          relatedFilms={relatedFilms}
        />
      </PageWrapper>
    );
  }
}

export {Movie};

const mapStateToProps: MapStateToProps<StateProps, OwnProps, State> = (state: State, ownProps: OwnProps): StateProps => ({
  films: getFilms(state),
  film: getFilm(state, ownProps.match.params.id),
  reviews: getReviews(state, ownProps.match.params.id),
  user: getUser(state)
});

const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = (dispatch: ThunkDispatch): DispatchProps => ({
  loadFilms: (): Promise<void> => dispatch(Operation.loadFilms()),
  loadReviews: (id: number): Promise<void> => dispatch(Operation.loadReviews(id))
});

const ConnectedComponent: any =
  connect<StateProps, DispatchProps, OwnProps, State>(mapStateToProps, mapDispatchToProps)(Movie as any);

export default ConnectedComponent;
