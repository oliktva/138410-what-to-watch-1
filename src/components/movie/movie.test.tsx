import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import {Movie} from './movie';
import {films, film} from 'src/fixtures/films';
import {reviews} from 'src/fixtures/reviews';
import {user} from 'src/fixtures/user';
import {match} from 'src/fixtures/router';

Enzyme.configure({adapter: new Adapter()});

it(`renders correctly`, () => {
  const movie = shallow(<Movie
    films={films}
    film={film}
    reviews={reviews}
    user={user}
    loadFilms={jest.fn()}
    loadReviews={jest.fn()}
    location={{} as any}
    history={{} as any}
    match={match}
  />);

  expect(toJson(movie)).toMatchSnapshot();
});
