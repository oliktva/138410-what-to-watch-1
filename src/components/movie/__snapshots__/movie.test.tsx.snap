// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<PageWrapper>
  <ConnectFunction
    film={
      Object {
        "backgroundColor": "#795433",
        "backgroundImage": "https://es31-server.appspot.com/wtw/static/film/background/Pulp_Fiction.jpg",
        "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
        "director": "Quentin Tarantino",
        "genre": "Crime",
        "id": 1,
        "isFavorite": false,
        "name": "Pulp Fiction",
        "posterImage": "https://es31-server.appspot.com/wtw/static/film/poster/Pulp_Fiction.jpg",
        "previewImage": "https://es31-server.appspot.com/wtw/static/film/preview/pulp-fiction.jpg",
        "previewVideoLink": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "rating": 8.9,
        "released": 1994,
        "runTime": 153,
        "scoresCount": 1635992,
        "starring": Array [
          "John Travolta",
          "Uma Thurman",
          "Samuel L. Jackson",
        ],
        "videoLink": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
      }
    }
    header={
      <Header
        className="movie-card__head"
        isLightLogo={true}
        user={
          Object {
            "avatarUrl": "img/1.png",
            "email": "Oliver.conner@gmail.com",
            "id": 1,
            "name": "Oliver.conner",
          }
        }
        withBreadcrumbs={false}
      />
    }
    relatedFilms={
      Array [
        Object {
          "backgroundColor": "#A6B7AC",
          "backgroundImage": "https://es31-server.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg",
          "description": "In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.",
          "director": "Martin Scorsese",
          "genre": "Crime",
          "id": 3,
          "isFavorite": false,
          "name": "Gangs of new york",
          "posterImage": "https://es31-server.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg",
          "previewImage": "https://es31-server.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg",
          "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
          "rating": 7.5,
          "released": 2002,
          "runTime": 167,
          "scoresCount": 370881,
          "starring": Array [
            "Leonardo DiCaprio",
            "Cameron Diaz",
            "Daniel Day-Lewis",
          ],
          "videoLink": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
        },
      ]
    }
    reviews={
      Array [
        Object {
          "comment": "This is comment",
          "date": "54367894",
          "id": 1,
          "rating": 8,
          "user": Object {
            "id": 2,
            "name": "Joe",
          },
        },
        Object {
          "comment": "This is comment",
          "date": "54367893",
          "id": 3,
          "rating": 7,
          "user": Object {
            "id": 2,
            "name": "Tatiana",
          },
        },
        Object {
          "comment": "This is comment",
          "date": "54367892",
          "id": 2,
          "rating": 6,
          "user": Object {
            "id": 2,
            "name": "Ivan",
          },
        },
      ]
    }
  />
</PageWrapper>
`;
